src/Microsoft.AspNetCore.Server.Kestrel.Core/Internal/ConnectionHandler.cs

```diff
@@ -44,7 +44,7 @@ public IConnectionContext OnConnection(IConnectionInformation connectionInfo)
                 ConnectionId = connectionId,
                 ServiceContext = _serviceContext,
                 PipeFactory = connectionInfo.PipeFactory,
-                ConnectionAdapters = connectionInfo.ListenOptions.ConnectionAdapters,
+                ConnectionAdapters = ((ListenOptions)connectionInfo.EndPointInformation).ConnectionAdapters,
```

Comment:
`// TODO: This is a hack, we need to associate kestrel specific things with endpoint information somehow`